packages/server/graphql/types/RetroDiscussStage.ts

```diff
@@ -14,6 +28,14 @@ const RetroDiscussStage = new GraphQLObjectType<any, GQLContext>({
   fields: () => ({
     ...newMeetingStageFields(),
     ...discussionThreadStageFields(),
+    discussion: {
+      type: GraphQLNonNull(Discussion),
+      description: 'The discussion about the stage or a dummy data when there is no disscussion',
+      resolve: async ({discussionId}, _args, {dataLoader}) => {
+        const discussion = await dataLoader.get('discussions').load(discussionId)
```

Comment:
the dataloader is gonna cause a hit to the database, which is avoidable if we know the discussionId is `''`.
can we do something like `return discussionId ? dataLoader.get(...) : DUMMY_DISCUSSION`

Reply:
that was my initial idea but... that's how 100% empty meeting looks like
```
{
  "id": "7",
  "type": "complete",
  "payload": {
    "data": {
      "viewer": {
        "newMeeting": {
          "__typename": "RetrospectiveMeeting",
          "id": "5d0pVmo7xC",
          "createdAt": "2021-07-14T14:42:28.197Z",
          "name": "Retro #1",
          "team": {
            "name": "bartosz’s Team",
            "id": "5coX2pqNRD"
          },
          "meetingType": "retrospective",
          "reflectionGroups": [],
          "meetingMembers": [
            {
              "tasks": [],
              "id": "local|5coX2kBCEw::5d0pVmo7xC",
              "__typename": "RetrospectiveMeetingMember",
              "user": {
                "rasterPicture": "https://action-files.parabol.co/static/avatars/ba.png",
                "preferredName": "bartosz",
                "id": "local|5coX2kBCEw"
              }
            }
          ],
          "phases": [
            {
              "__typename": "CheckInPhase",
              "phaseType": "checkin",
              "id": "5d0pVoEUA9"
            },
            {
              "__typename": "ReflectPhase",
              "phaseType": "reflect",
              "id": "5d0pVqmvmw"
            },
            {
              "__typename": "GenericMeetingPhase",
              "phaseType": "group",
              "id": "5d0pVqmvmz"
            },
            {
              "__typename": "GenericMeetingPhase",
              "phaseType": "vote",
              "id": "5d0pVqmvmB"
            },
            {
              "__typename": "DiscussPhase",
              "phaseType": "discuss",
              "stages": [
                {
                  "id": "5d0pVqmvmF",
                  "reflectionGroupId": "",
                  "reflectionGroup": {
                    "title": "Unknown",
                    "voteCount": 0,
                    "reflections": [],
                    "id": "5d0pVmo7xC:dummyGroup"
                  },
                  "discussion": {
                    "commentCount": 0,
                    "id": "5d0pVqmvmG"
                  }
                }
              ],
              "id": "5d0pVqmvmD"
            }
          ]
        },
        "id": "local|5coX2kBCEw"
      }
    }
  }
}
```
I'll dig into this and see if where how discussionId is created.

Reply:
This `discussionId` is somehow taken from rethinkdb, but honestly, I can't find which resolver fetches that 🙈

Reply:
Ohhhh that's right!
Ugh, placeholders are just the worst.
Basically, we create that dummy with a UID.
Then, when the real data comes in, we swap out the dummy date for the new data but keep the UID the same.
You can see the stage getting created here: https://github.com/ParabolInc/parabol/blob/ba2fdeadc7da023f188e2765096810ebdaf59ccd/packages/server/database/types/DiscussPhase.ts
and if you check out the DiscussStage constructor, you'll see it just creates a new UID.

So how do we now it's dummy data?

```ts
const isDummy = stages[0].length === 1 && stages[0].reflectionGroupId === ''
return isDummy ? DUMMY_DATA : dataLoader.get(...)`
```

Reply:
Thank you, Matt! In this particular resolver, we can use `reflectionGroupId` directly so it's even easier. 
```
      resolve: async ({discussionId, reflectionGroupId}, _args, {dataLoader}) => {
        const isDummy = reflectionGroupId === ''
```
Works like a charm!